app/assets/javascripts/admin/addon/routes/admin.js

```diff
@@ -1,19 +1,31 @@
+import { tracked } from "@glimmer/tracking";
 import { inject as service } from "@ember/service";
-import { ADMIN_PANEL, MAIN_PANEL } from "discourse/lib/sidebar/panels";
+import {
+  ADMIN_PANEL,
+  COMBINED_MODE,
+  MAIN_PANEL,
+  SEPARATED_MODE,
+} from "discourse/lib/sidebar/panels";
 import DiscourseRoute from "discourse/routes/discourse";
 import I18n from "discourse-i18n";
 
 export default class AdminRoute extends DiscourseRoute {
   @service sidebarState;
   @service siteSettings;
   @service currentUser;
+  @tracked initialSidebarState;
 
   titleToken() {
     return I18n.t("admin_title");
   }
 
   activate() {
     if (this.currentUser.use_admin_sidebar) {
+      this.initialSidebarState = {
+        mode: this.sidebarState.mode,
+        displaySwitchPanelButtons: this.sidebarState.displaySwitchPanelButtons,
+      };
+
```

Comment:
I actually wonder if the `sidebarState` service can track the "last" state and then API users can just call `revert` to previous state when they want to navigate away. 

cc @lis2

Reply:
Interesting concept. So far we have 3 panels, "main", "chat" and "admin" but assume that in the future we will have more. We can remember the last mode per panel and as you suggested keep it in `sidebarState`.

Each time we change panel we can check if there is a history state first for the new panel.

Reply:
Yes that could be another way of approaching it too :+1: We haven't really needed it in other situations until now.

Reply:
Though, I would at least like to get this fix out then we can follow up with a nicer way of doing it.

Reply:
> we can follow up with a nicer way of doing it.

Absolutely 👍 Does @lis2 have the capacity to do this? 😁